### DISCUSSION TITLE
Service Tags and IP ranges for Codespaces

### DISCUSSION THREAD

**Discussion by fabianrmendez** [community/community#11309] [Codespaces] [answered]
We have a web app that connects to Azure SQL and Azure Data Explorer databases. The IP for each session in Codespaces is dynamic and this causes issues when trying to whitelist the IPs. In the case of ADX, its inside a virtual network and in order to allow inbound traffic from Codespaces, we have to manually update the network security rules for each session. This could be avoided if there was a service tag for Codespaces. For the SQL database, we have to manually update the firewall rules. In this case there are no service tags that can be used, but it would make things cleaner if there was a defined range of IPs dedicated to Codespaces (the published IP range for GitHub is so large that is not practical to whitelist all of it).

Are these features something that is currently in the plans? Or is there an alternative to reroute Codespaces traffic through a NAT Gateway or similar so we can whitelist this instead?

**Comment by tanmayeekamath** [comment, marked as answer]
Hey @fabianrmendez we have plans to add support for Azure VNET integration that would enable spinning up Codespaces in your Azure VNET so that those instances can then access any resources within your networking topology and follow your networking policies/firewall rules etc. 

Would this address your problem?

**Reply by srozanc-mcs** [reply]
@tanmayeekamath This feature would be amazing! Do you have any update for when this would be coming?

**Reply by hfhbd** [reply]
Do you have any news? https://github.com/github/roadmap/issues/534 was closed because it was "shipped", but there are no new information/docs or a changelog entry.

**Comment by callain** [comment]
Hello, we had the same issue. I created a script which automatically add the codespace IP in the Azure MySQL firewall.
It's just too bad that I can't remove it when deleting the codespace.
We will switch to "Azure VNET integration" when it will be ready then 👍 

```bash
  az login --service-principal --username $AZURE_CODESPACE_APP_ID --password $AZURE_CODESPACE_APP_PASSWORD --tenant $AZURE_TENANT_ID
  CODESPACE_IP=`curl ifconfig.me`
  az mysql server firewall-rule create --resource-group $AZURE_RESOURCE_GROUP --name $CODESPACE_NAME --server-name $JDBC_HOSTNAME --start-ip-address $CODESPACE_IP --end-ip-address $CODESPACE_IP
  ```

You can configure it as postCreateCommand.
